### ISSUE TITLE
Security: HASS.Agent Installs vulnerable driver

### ISSUE THREAD

**Issue by leankuroneko** [hass-agent/HASS.Agent#163]
**Describe the bug**
After installing HASS.Agent, ESET Endpoint Security raised an alert and quarantined the file `inpoutx64.sys` for being _potentially unsafe_

Searching about this I found in the [ESET Forum](https://forum.eset.com/topic/40479-inpoutx64sys-potentially-dangerous-application/) that although [not infected](https://www.virustotal.com/gui/file/f8965fdce668692c3785afa3559159f9a18287bc0d53abb21902895a8ecf221b/detection) this file [has vulnerabilities](https://www.loldrivers.io/drivers/91ff1575-9ff2-46fd-8bfe-0bb3e3457b7f/
) that can be used in [Privilege Escalation](https://news.sophos.com/en-us/2022/10/04/blackbyte-ransomware-returns/). This concern was also raised in [MalwareTips](https://malwaretips.com/threads/is-inpoutx64-sys-safe.132331/) and in a [Windows Community post](https://answers.microsoft.com/en-us/windows/forum/all/what-is-inpoutx64sys-and-is-it-needed/7c31b240-2df1-439e-ac06-ccfc873aa9a7) where it's mentioned to be blocked by Riot's Anti Cheat Engine.

> Maintainer: _Personally I haven't used the thing myself for about 7 years_
>  https://forums.highrez.co.uk/viewtopic.php?p=17167&sid=c775cbad2219955de63ce2821f17cbd5#p17167

> Footer: _The Author makes no guarantee that this software is free from bugs and will not harm your system._
> https://highrez.co.uk/

I'd suggest its removal or replacement since given it seems abandoned

**Screenshots**
![image](https://github.com/user-attachments/assets/16b50b91-7028-4709-a88e-1bc43ca974a6)
> Scan Module: Real-Time File System Protection
> Object: C:\WINDOWS\system32\Drivers\inpoutx64.sys
> Detection: Win64/HighRez.A (Potentially Unsafe Application)
> User: NT AUTHORITY\SYSTEM
> Information: An event occurred in a new file created by the application: C:\Program Files\HASS.Agent\Service\HASS.AgentSatelliteService.exe (4E72406DE7447604BB86085F5DDAFAB8BAOB57C5).
> Hash: 6AFC6B04CF73DD461E4A4956365F25C1F1162387

**Misc info (please complete the following information):**
 - Windows build: 23H2 (22631.4169)
 - Windows' UI language: es_AR
 - HASS.Agent version: 2.0.1
- ESET Endpoint Security: 11.1.2039.2 (Detection engine: [29912](https://go.eset.com/virushistory?lng=3082&product=ees&version=11.1))

Please check what's applicable (multiple answers possible):

 - [x] Installed via installer
 - [x] Problem occurs in HASS.Agent

**Comment by amadeo-alex** [comment]
Thank you for the information! [Looks like](https://github.com/hass-agent/HASS.Agent/issues/43) the next release after 2.1.0 will be a security themed one.
I'll have quite more time next week to work on HASS.Agent so I'll take a deeper dive to investigate this.

**Comment by patrickdickey52761** [comment]
I'm having a similar issue with the Satellite Service driver. As soon as I finished the installation, Avast blocked the driver. On their end, the only way to allow the driver is to turn off the protection completely (which is a bug in their program, IMHO). Here's a screenshot of the warning message.  I'm adding it to this as opposed to creating a new issue because it's related to the underlying issue of potentially unsafe drivers/code being used.

Have a great day. :)
Patrick.
![AvastWarningDriver](https://github.com/user-attachments/assets/d50dcb51-8faf-4b25-a86a-164b833e15fd)

**Comment by loryanstrant** [comment]
Similar problem for me, my org has blocked it via Windows Defender:

![Image](https://github.com/user-attachments/assets/fa6f0084-345c-41dc-81fe-77cac9cd8151)

**Comment by l3ztum** [comment]
![Image](https://github.com/user-attachments/assets/b289b3b1-024f-4deb-a556-d079b46d3bcd) Windows Defender also pinged me about the Satellite service. Flagged as "HackTool:Win32/Winring0"

**Comment by sdennis91** [comment]
This guy is using sams idea to install viruses on our computers. I have confirmed. Im going back  to sams version and this guys fork should be removed from github.

**Comment by amadeo-alex** [comment]
Closed with 2.2.0 release, please feel free to create a new issue if required and reference this one.
